x/auxiliary/types/msg_batch.go

```diff
@@ -0,0 +1,102 @@
+package types
+
+import (
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/axelarnetwork/utils/funcs"
+	"github.com/axelarnetwork/utils/slices"
+)
+
+var (
+	_ sdk.Msg = &BatchRequest{}
+
+	_ cdctypes.UnpackInterfacesMessage = &BatchRequest{}
+)
+
+// NewBatchRequest is the constructor for BatchRequest
+func NewBatchRequest(sender sdk.AccAddress, messages []sdk.Msg) *BatchRequest {
+	return &BatchRequest{
+		Sender:   sender,
+		Messages: slices.Map(messages, func(msg sdk.Msg) cdctypes.Any { return *funcs.Must(cdctypes.NewAnyWithValue(msg)) }),
+	}
+}
+
+// ValidateBasic executes a stateless message validation
+func (m BatchRequest) ValidateBasic() error {
+	if err := sdk.VerifyAddressFormat(m.Sender); err != nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, sdkerrors.Wrap(err, "sender").Error())
+	}
+
+	if len(m.Messages) == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "empty batch")
+	}
+
+	if AnyBatch(m.UnwrapMessages()) {
```

Comment:
Add a unit test for this so that `BatchRequest` won't become nested accidentally.